machine.go

```diff
@@ -590,6 +590,20 @@ func captureFifoToFile(logger *log.Entry, fifoPath string, fifo io.Writer) error
 
 	logger.Debugf("Capturing %q to writer", fifoPath)
 
+	// this goroutine is to track the life of the application along with whether
+	// or not the context has been cancelled. In the event that the context has
+	// been cancel or the application has closed, we will close the fifo file
+	go func() {
+		select {
+		case <-ctx.Done():
```

Comment:
Is there a reason to also use the `ctx` here instead of just using `m.exitCh` exclusively? IIUC the `ctx` passed here will ultimately [be passed in via `machine.Start` here](https://github.com/firecracker-microvm/firecracker-go-sdk/blob/437f90243b6d40db06333829a6dc2c38e7570af4/machine.go#L308), which is also the same [`ctx` passed to `startVMM`](https://github.com/firecracker-microvm/firecracker-go-sdk/blob/437f90243b6d40db06333829a6dc2c38e7570af4/handlers.go#L118), which in turn is [a `ctx` used to determine when to close `exitCh`](https://github.com/firecracker-microvm/firecracker-go-sdk/blob/437f90243b6d40db06333829a6dc2c38e7570af4/machine.go#L503-L514).

It doesn't break anything to use `ctx` here but it does increase the complexity for readers to go through to determine when this select will get unblocked, so all else being equal I'd prefer we only use `exitCh`.

Reply:
Sure, fixed. They have to go through that logic anyway to determine that `exitCh` is closed when the context is closed. So, I don't think it adds anymore complexity then what they already have to do, however.